### PR TITLE
Add IDeadLetterMessageRepository

### LINKED ISSUE #10: Add DLMRepository - mock

### PR DESCRIPTION
**Pull request by matee911**
#10

### CODE REVIEW

File: src/ClayPipe/Models/DeadLetterMessage.cs [hunk 1]
```diff
@@ -20,4 +20,14 @@ public class DeadLetterMessage
...
     }
 
+    public interface IDeadLetterMessageRepository
+    {
+        void Ingest();
+        IEnumerable<DeadLetterMessage> FindMessages(string queueName, int perPage, int page);
+        DeadLetterMessage Get(string id);
+        void Update(DeadLetterMessage dlm);
+        void Delete(DeadLetterMessage dlm);
+        void Resend(DeadLetterMessage dlm);
```
Comment (matee911) [inline src/ClayPipe/Models/DeadLetterMessage.cs]: @ekamil  @gacu82  Maybe we should use here some enums instead of voids or maybe something else?

**Comment by matee911** [comment]
@ekamil @gacu82 guys are you there?

**Comment by matee911** [comment]
Let's collaborate they said,
it will go faster they said...

File: src/ClayPipe/Models/DeadLetterMessage.cs (hunk 1 above)
Comment (ekamil) [inline src/ClayPipe/Models/DeadLetterMessage.cs]: Let's just use exceptions (but we need to keep them in docstrings). Void is ok for "everything's fine".

**Pull request merged**